### PR TITLE
feat: KCP AWS VpcPeering load remote vpc

### PR DESCRIPTION
**Pull request by vandjelk**
**Description**

Changes proposed in this pull request:

- Add provider to state so that remote vpc can be loaded
- load remote vpc into state
- create vpc peering connection mocks
- integration tests updated

### CODE REVIEW

File: internal/controller/cloud-control/vpcpeering_aws_test.go [hunk 1]
```diff
@@ -63,6 +73,14 @@ var _ = Describe("Feature: KCP VpcPeering", func() {
 				).
 				Should(Succeed())
 		})
+
+		By("And Then KCP VpcPeering has status.vpcId equal to existing AWS VPC id", func() {
+			Expect(vpcpeering.Status.VpcId).To(Equal(vpcId))
+		})
+
+		By("And Then KCP VpcPeering has status.ConnectionId equal to existing AWS Connection id", func() {
+			Expect(vpcpeering.Status.ConnectionId).To(Equal(connectionId))
```
Comment (tmilos77) [inline internal/controller/cloud-control/vpcpeering_aws_test.go:82]: Shouldn't it be listed from the mock and checked if mock value is equal to the status value, rather then be defined so hard-coded in the test?

File: internal/controller/cloud-control/vpcpeering_aws_test.go [hunk 2]
```diff
@@ -19,6 +19,7 @@ var _ = Describe("Feature: KCP VpcPeering", func() {
 			remoteVpcId     = "6e6d1748-9912-4957-9075-b97a6fac8ac1"
 			remoteAccountId = "123"
 			remoteRegion    = "eu-west1"
+			connectionId    = "26ce833e-07d1-4493-98ee-f9d6f11a6987->6e6d1748-9912-4957-9075-b97a6fac8ac1"
```
Comment (tmilos77) [inline internal/controller/cloud-control/vpcpeering_aws_test.go:22]: Is AWS peering connection id really in this format?

File: pkg/kcp/provider/aws/vpcpeering/client/client.go [hunk 3]
```diff
@@ -34,3 +35,20 @@ func (c *client) DescribeVpcs(ctx context.Context) ([]types.Vpc, error) {
 	}
 	return out.Vpcs, nil
 }
+
+func (c *client) CreateVpcPeeringConnection(ctx context.Context, vpcId, remoteVpcId, remoteRegion, remoteAccountId *string) (*types.VpcPeeringConnection, error) {
+	//out, err := c.svc.CreateVpcPeeringConnection(ctx, &ec2.CreateVpcPeeringConnectionInput{
```
Comment (tmilos77) [inline pkg/kcp/provider/aws/vpcpeering/client/client.go:40]: Code comments should not be committed

File: pkg/kcp/provider/aws/vpcpeering/loadRemoteVpc.go [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+	if err != nil {
+		return composed.LogErrorAndReturn(err, "Error loading AWS VPC Networks", composed.StopWithRequeue, ctx)
+	}
+
+	var vpc *ec2Types.Vpc
+	var remoteVpcName string
+	var allLoadedVpcs []string
+	// TODO refactor as it is more or less the same as in loadVpc
+
+	for _, v := range vpcList {
```
Comment (tmilos77) [inline pkg/kcp/provider/aws/vpcpeering/loadRemoteVpc.go:56]: ❗  Make a copy of `v` before taking it's ref...
```
for _, vv := range vpcList {
    v := vv
   vpc = &v
}
```

File: pkg/kcp/provider/aws/vpcpeering/new.go [hunk 5]
```diff
@@ -26,6 +26,8 @@ func New(stateFactory StateFactory) composed.Action {
 				composed.ComposeActions("awsVpcPeering-non-delete",
 					addFinalizer,
 					loadVpc,
+					loadRemoteVpc,
+					createVpcPeeringConnection,
```
Comment (tmilos77) [inline pkg/kcp/provider/aws/vpcpeering/new.go:30]: Before a peering connection is made, it first should be checked if one already exists

File: pkg/kcp/provider/aws/vpcpeering/updateSuccessStatus.go [hunk 6]
```diff
@@ -4,26 +4,19 @@ import (
...
-
-	return nil, nil
+	return composed.UpdateStatus(state.ObjAsVpcPeering()).
+		SetCondition(metav1.Condition{
+			Type:    cloudcontrol1beta1.ConditionTypeReady,
+			Status:  "True",
+			Reason:  cloudcontrol1beta1.ReasonReady,
+			Message: "Additional VpcPeerings(s) are provisioned",
+		}).
+		ErrorLogMessage("Error updating VpcPeering success status").
```
Comment (tmilos77) [inline pkg/kcp/provider/aws/vpcpeering/updateSuccessStatus.go:20]: Without specifying `SuccessError()` default is `composed.StopAndForget`, hope that's intended.

File: internal/controller/cloud-control/vpcpeering_aws_test.go (hunk 2 above)
Comment (vandjelk) [reply internal/controller/cloud-control/vpcpeering_aws_test.go:22]: No. It is actually something like pcx-111aaa111 but this was a quick way for me to create unique identifier between two vpcs.

File: internal/controller/cloud-control/vpcpeering_aws_test.go (hunk 1 above)
Comment (vandjelk) [reply internal/controller/cloud-control/vpcpeering_aws_test.go:82]: I agree. I will use values from mock.

**Review by tmilos77 — APPROVED** [review]

**Review by tmilos77 — APPROVED** [review]

**Pull request merged**